riscv/mmu.h

```diff
@@ -351,7 +351,7 @@ class mmu_t
 
   int is_misaligned_enabled()
   {
-    return proc && proc->get_cfg().misaligned;
+    return proc && proc->extension_enabled(EXT_ZICCLSM);
```

Comment:
does it make sense to add `private` member `bool processor_t::is_misaligned_enabled` that we don't check this every time is called? 
`is_misaligned_enabled` should be initialized in constructor body

Reply:
I think it is OK as-is.  `is_misaligned_enabled` is not on a critical performance path, and it is not very expensive to evaluate.